packages/Combobox/src/combobox.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  chakra,
+  omitThemingProps,
+  PropsOf,
+  StylesProvider,
+  ThemingProps,
+  useMultiStyleConfig,
+  useStyles,
+} from "@chakra-ui/system"
+import { InputProps } from "@chakra-ui/input"
+import {
+  cx,
+  mergeRefs,
+  ReactNodeOrRenderProp,
+  __DEV__,
+  runIfFn,
+} from "@chakra-ui/utils"
+import * as React from "react"
+import {
+  ComboboxContextProvider,
+  useCombobox,
+  UseComboboxReturn,
+  UseComboboxProps,
+  useComboboxInput,
+  UseComboboxInputProps,
+  useComboboxList,
+  useComboboxListItem,
+} from "./combobox.hook"
+
+export type ComboboxProps = UseComboboxProps &
+  ThemingProps & {
+    children: ReactNodeOrRenderProp<UseComboboxReturn>
+  }
+
+/**
+ * The wrapper component that provides context, state, and focus
+ * management to its sub-components.
+ *
+ * It doesn't render any DOM node.
+ */
+export function Combobox(props: ComboboxProps) {
+  const styles = useMultiStyleConfig("Combobox", props)
+  const realProps = omitThemingProps(props)
+  const comboboxCtx = useCombobox(realProps)
+  const context = React.useMemo(() => comboboxCtx, [comboboxCtx])
+  return (
+    <ComboboxContextProvider value={context}>
+      <StylesProvider value={styles}>
+        {runIfFn(props.children, context)}
+      </StylesProvider>
+    </ComboboxContextProvider>
+  )
+}
+
+if (__DEV__) {
+  Combobox.displayName = "Combobox"
+}
+
+export type ComboboxInputProps = PropsOf<typeof chakra.input> &
+  UseComboboxInputProps &
+  InputProps
+
+export const ComboboxInput = React.forwardRef(function StyledComboboxInput(
```

Comment:
The `Combobox` component has a multi part style object. I couldn't find a way to create a nested multi part style object.

For example the `Combobox` will have an `combobox-input`, `combobox-list` and `combobox-list-item` style parts.  
But the `combobox-input` will also have a `field` and an `addon` style part.  

Is there a way to create a nested multi part theme style?  
Should I create an additional multi part style object called `ComboboxInput` that inherit the styles from the input and additional combobox input specific styles?  
Or is there any other way to do this that I am not seeing?

Reply:
Yes, there's a way to do this @leoalves.

I'd recommend you check the theme package to see examples of how to define multipart theme styles.

Thank you for working on this 💖

Reply:
Based on how to set up the theming, here are some steps:

- Add `combobox.ts` to `packages/theme/src/components`
- Create a theme structure that looks like this:

```ts
const parts = {
  list: "the group of combobox items",
  input: "the combobox input",
  item: "the combobox item or option"
}

const baseStyle = {
  input: {},
  list: {},
  item: {}
}

const combobox = multiStyleConfig({
 parts,
  baseStyle,
 // you can add sizes, and variants as well
})

export default combobox
```

- Add combobox to the theme's index file
- Build the theme package `yarn theme build`

That's it! Let me know if you have any questions